bzip2.rs

```diff
@@ -1077,6 +1118,24 @@ unsafe fn hasSuffix(s: *mut c_char, suffix: *const c_char) -> Bool {
     0 as Bool
 }
 
+#[cfg(windows)]
+/// Prevent Windows from mangling the read data.
+unsafe fn set_binary_mode(file: *mut FILE) {
+    use std::ffi::c_int;
+
+    extern "C" {
+        fn _setmode(fd: c_int, mode: c_int) -> c_int;
+    }
+
+    if _setmode(fileno(file), 0x8000 /* O_BINARY */) == -1 {
```

Comment:
that `0x8000` is not a libc symbol?

Reply:
I was looking at the linux docs for libc which doesn't have this. On Windows it does exist in the libc crate however.